zcash_client_backend/src/wallet.rs

```diff
@@ -65,15 +65,15 @@ impl NoteId {
 /// internal account ID and the pool to which funds were sent in the case of a wallet-internal
 /// output.
 #[derive(Debug, Clone)]
-pub enum Recipient<N> {
+pub enum Recipient<AccountId, N> {
     Transparent(TransparentAddress),
     Sapling(sapling::PaymentAddress),
     Unified(UnifiedAddress, PoolType),
     InternalAccount(AccountId, N),
 }
 
-impl<N> Recipient<N> {
-    pub fn map_internal_account<B, F: FnOnce(N) -> B>(self, f: F) -> Recipient<B> {
+impl<AccountId, N> Recipient<AccountId, N> {
+    pub fn map_internal_note<B, F: FnOnce(N) -> B>(self, f: F) -> Recipient<AccountId, B> {
```

Comment:
It's still an internal account, however; this rename is just because it's only the note part of the `InternalAccount` variant that is being mapped over. While fundamentally this could still just be named `map`, `map_internal_account_note` would be better.